aiida/calculations/transfer.py

```diff
@@ -0,0 +1,250 @@
+            if computer.name != node_object.computer.name:
+                error_message = ' > remote node `{}` points to computer `{}`, not the one being used (`{}`)'
+                error_message = error_message.format(node_label, node_object.computer.name, computer.name)
+                error_message_list.append(error_message)
+
+    for source_label, _, _ in local_files:
+        source_nodes_required.add(source_label)
+        source_node = source_nodes.get(source_label, None)
+        error_message = check_node_type('local_files', source_label, source_node, orm.FolderData)
+        if error_message:
+            error_message_list.append(error_message)
+
+    for source_label, _, _ in remote_files:
+        source_nodes_required.add(source_label)
+        source_node = source_nodes.get(source_label, None)
+        error_message = check_node_type('remote_files', source_label, source_node, orm.RemoteData)
+        if error_message:
+            error_message_list.append(error_message)
+
+    for source_label, _, _ in symlink_files:
+        source_nodes_required.add(source_label)
+        source_node = source_nodes.get(source_label, None)
+        error_message = check_node_type('symlink_files', source_label, source_node, orm.RemoteData)
+        if error_message:
+            error_message_list.append(error_message)
+
+    unrequired_nodes = source_nodes_provided.difference(source_nodes_required)
+    for node_label in unrequired_nodes:
+        error_message = ' > node `{}` provided as inputs is not being used'
+        error_message = error_message.format(node_label)
+        error_message_list.append(error_message)
+
+    if len(error_message_list) > 0:
+        error_message = '\n\n'
+        for error_add in error_message_list:
+            error_message = error_message + error_add + '\n'
+        return error_message
+
+
+def check_node_type(list_name, node_label, node_object, node_type):
+    """Common utility function to check the type of a node"""
+
+    if node_object is None:
+        error_message = ' > node `{}` requested on list `{}` not found among inputs'
+        return error_message.format(node_label, list_name)
+
+    if not isinstance(node_object, node_type):
+        error_message = ' > node `{}`, requested on list {} should be of type `{}`'
+        return error_message.format(node_label, list_name, node_type.class_node_type)
+
+    return None
+
+
+class TransferCalculation(CalcJob):
+    """Utility to copy files from different FolderData and RemoteData nodes into a single place.
+
+    The final destination for these files can be either the local repository (by creating a
+    new FolderData node to store them) or in the remote computer (by leaving the files in a
+    new remote folder saved in a RemoteData node).
+
+    Only files from the local computer and a single remote computer can be moved with a single
+    instance of this CalcJob (so you can't directly move files from remote to remote and all
```

Comment:
This isn't fully clear. When you say "you can't directly move files from remote to remote" that is not really true, because you can move files from a remote folder to another remote folder, right? Only restriction is just that you cannot transfer between different remote computers. So I would simplify this entire paragraph, you can move files between remote folders as long as they are on the same computer.